scripts/init.sh

```diff
@@ -3,92 +3,111 @@
 
 set -e
 
-case "$(uname -s):$(uname -m):$(uname -v):$(uname -a)" in
-Linux:x86_64*)     
-    TARGET="x86_64-unknown-linux-musl"
-    ;;
-Linux:aarch64:*:*Android*)
-    TARGET="aarch64-linux-android"
-    ;;
-Linux:arm64*|Linux:aarch64*)
-    TARGET="aarch64-unknown-linux-musl"
-    ;;
-Linux:arm*)     
-    TARGET="armv7-unknown-linux-musleabihf"
-    ;;
-Linux:i686:iSH*)
-    TARGET="ish"
-    ;;
-Linux:i686*)     
-    TARGET="i686-unknown-linux-musl"
-    ;;
-Linux:i586*)     
-    TARGET="i586-unknown-linux-musl"
-    ;;
-Linux:mips*)     
-    TARGET="mips-unknown-linux-musl"
-    ;;
-FreeBSD:x86_64*)
-    TARGET="x86_64-unknown-freebsd"
-    ;;
-FreeBSD:amd64*)
-    TARGET="x86_64-unknown-freebsd"
-    ;;
-FreeBSD:i686*)
-    TARGET="i686-unknown-freebsd"
-    ;;
-Darwin:x86_64*)    
-    TARGET="x86_64-apple-darwin"
-    ;;
-WindowsNT:x86_64*)    
-    TARGET="x86_64-pc-windows-msvc.exe"
-    ;;
-WindowsNT:i686*)    
-    TARGET="i686-pc-windows-msvc.exe"
-    ;;
-*)          
-    echo "Unsupported system ($(uname -a))"
-    exit 1
-    ;;
-esac
+main() {
+    case "$(uname -s):$(uname -m):$(uname -v):$(uname -a)" in
+    Linux:x86_64*)     
+        TARGET="x86_64-unknown-linux-musl"
+        ;;
+    Linux:aarch64:*:*Android*)
+        TARGET="aarch64-linux-android"
+        ;;
+    Linux:arm64*|Linux:aarch64*)
+        TARGET="aarch64-unknown-linux-musl"
+        ;;
+    Linux:arm*)     
+        TARGET="armv7-unknown-linux-musleabihf"
+        ;;
+    Linux:i686:iSH*)
+        TARGET="ish"
+        ;;
+    Linux:i686*)     
+        TARGET="i686-unknown-linux-musl"
+        ;;
+    Linux:i586*)     
+        TARGET="i586-unknown-linux-musl"
+        ;;
+    Linux:mips*)     
+        TARGET="mips-unknown-linux-musl"
+        ;;
+    FreeBSD:x86_64*)
+        TARGET="x86_64-unknown-freebsd"
+        ;;
+    FreeBSD:amd64*)
+        TARGET="x86_64-unknown-freebsd"
+        ;;
+    FreeBSD:i686*)
+        TARGET="i686-unknown-freebsd"
+        ;;
+    Darwin:x86_64*)    
+        TARGET="x86_64-apple-darwin"
+        ;;
+    WindowsNT:x86_64*)    
+        TARGET="x86_64-pc-windows-msvc.exe"
+        ;;
+    WindowsNT:i686*)    
+        TARGET="i686-pc-windows-msvc.exe"
+        ;;
+    *)          
+        echo "Unsupported system ($(uname -a))"
+        exit 1
+        ;;
+    esac
 
-if [ -z "$TMPDIR" ]
-then
-    TMPDIR="/tmp"
-fi
-
-TEMP_PATH="$TMPDIR/tunshell"
-CLIENT_PATH="$TEMP_PATH/client"
-HEADERS_PATH="$TEMP_PATH/headers"
+    if [ ! -z "$XDG_CACHE_HOME" ]
+    then
+        TEMP_PATH="$XDG_CACHE_HOME"
+    elif [ ! -z "$TMPDIR" ]
+    then
+        TEMP_PATH="$TMPDIR"
+    elif [ -x "$(command -v mktemp)" ]
+    then
+        TEMP_PATH="$(mktemp -d)"
```

Comment:
This will create a new directory every time, meaning that you will never hit your cache. This would probably be fine if you updated the `$XDG_CACHE_HOME` code to use `$HOME/.cache` if `$HOME` exists as workstations would generally use that case and things like docker containers don't get any value from the cache anyways as their filesystem is generally thrown away each invocation.

However in this case you will want to delete the directory at the end of the script so that you don't fill up users' filesystems.

Reply:
See above comment

Reply:
Updated the script to use `$HOME/.cache` ahead of tmp dirs